tests/unit/array/test_sequence.py

```diff
@@ -260,3 +262,37 @@ def test_set_and_append(index, storage, config):
         da.append(Document(id='new_new'))
 
     assert da[:, 'id'] == ['0', 'new', '2', '3', '4', 'new_new']
+
+
+@pytest.mark.parametrize(
+    'da_cls,config',
+    [
+        (DocumentArrayInMemory, lambda: None),
+        (DocumentArraySqlite, lambda: None),
+        # DocArrays backed by Weaviate in 0.17.0 has a refcount bug
+        # Remove this comment when the bug is fixed
+        # (DocumentArrayWeaviate, lambda: WeaviateConfig(n_dim=3)),
+        (DocumentArrayQdrant, lambda: QdrantConfig(n_dim=3)),
+        (DocumentArrayElastic, lambda: ElasticConfig(n_dim=3)),
+        (DocumentArrayRedis, lambda: RedisConfig(n_dim=3)),
+    ],
+)
+def test_deletion(da_cls, config, start_storage, capsys):
+    # Mock Deletion
+    class MockDA(da_cls):
+        def __del__(self):
+            assert not self._deleted
+            print('Goodbye')
+            self._deleted = True
+
+    da = MockDA(config=config())
+    assert not da._deleted
+    assert (
+        sys.getrefcount(da) == 2
+    ), f'{da_cls.__name__} initialized with {sys.getrefcount(da) - 1} references'
+
+    del da
+    gc.collect()
```

Comment:
I am not sure this is guarantee of anything

Reply:
I do not like having a test relying on this

Reply:
But I don't think there is a nice way to test the module exceptions. What we are looking for is a way to capture behavior of a SystemExit.

I have thought of some janky ways to do it though.

One such janky way is to have the test spawn a subprocess and assert that the cleanup of the subprocess doesn't raise any errors. It's kind of hacky so I'm still looking for a nicer way. But this might be the only way to go about it.

Reply:
We could also assert that the DocArray created by the subprocess can be loaded correctly.

But yea, using subprocess in tests seems a bit hacky